src/grammar/ast.ts

```diff
@@ -320,6 +321,14 @@ export type AstStatementForEach = {
     loc: SrcInfo;
 };
 
+export type AstStatementDestruct = {
+    kind: "statement_destruct";
+    identifiers: AstDestructMapping[];
```

Comment:
Let's use a `Map` here, this will simplify some part of the code, like the codegen

Reply:
done